doc/user_guide/big_data.rst

```diff
@@ -0,0 +1,228 @@
+^^^^^^^^^^^^^^^^^^^^^^^
+
+To convert a regular HyperSpy signal to a lazy one (so that any future
+operations are only performed lazily) we added
+:py:meth:`~.signal.BaseSignal.as_lazy` method:
+
+.. code-block:: python
+
+    >>> s = hs.signals.Signal1D(np.arange(150.).reshape((3, 50)))
+    >>> s
+    <Signal1D, title: , dimensions: (3|50)>
+    >>> sl = s.as_lazy()
+    >>> sl
+    <LazySignal1D, title: , dimensions: (3|50)>
+
+
+
+Constraints
+-----------
+
+A couple of constraints to ``LazySignal`` when compared to normal HyperSpy
+signals.
+
+Immutable signals
+^^^^^^^^^^^^^^^^^
+
+One of the most often encountered constraint when using ``LazySignal`` is the
+inability to modify an existing signal data (immutability). It is a logical
+consequence of the DAG (tree structure), where a complete history of the
+processing has to be stored to traverse later. 
+
+In fact, ``LazySignal`` removes the need for such operation, since only
+additional tree branches are added, requiring very little resources. In
+practical terms:
+
+.. code-block:: python
+
+    >>> s = s + 1 # instead of s += 1
+    >>> # or, even better
+    >>> s1 = s + 1
+
+Machine learning (decomposition)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:ref:`Machine learning<ml>` often performs large matrix manipulations,
+requiring significantly more memory than just storing the data. Lazy HyperSpy
+signals attempted to provide similar alternatives. These algorithms read the
+data in an "online" manner (i.e. only loading each element once or twice),
+enabling performing the decompositions on very large datasets. All of the
+algorithms are only approximations to the original implementations, some
+(ONMF in particular) sparsely tested.
+
+To facilitate the usual HyperSpy workflows as well as possible,
+:py:meth:`~._signals.lazy.LazySignal.decomposition` method of the
+``LazySignal`` offers implementations of:
+
+* **PCA** (``algorithm='PCA'``): performs the
+  `IncrementalPCA <http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.IncrementalPCA.html#sklearn.decomposition.IncrementalPCA>`_
+  from ``scikit-learn``.
+* **NMF** (``algorithm='ONMF'``): runs Online Robust NMF, as per "OPGD"
+  algorithm in `this paper by Zhao et. al
+  <http://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=7472160&isnumber=7471614>`_.
+* **ORPCA** (``algorithm='ORPCA'``): runs Online Robust PCA, is also available
+  for regular signals
+
+Minor changes
+^^^^^^^^^^^^^
+
+* **Histograms** for a ``LazySignal`` do not support ``knuth`` and ``blocks``
+  binning algorithms.
+* **CircleROI** sets the elements outside the ROI to ``np.nan`` instead of
+  using a masked array, because ``dask`` does not support masking. As a
+  convenience, ``nansum``, ``nanmean`` and other ``nan*`` signal methods were
+  added to mimic the workflow as closely as possible.
+
+
+Data processing with LazySignal
+-------------------------------
+
+Despite the aforementioned constraints, most of the usual HyperSpy operations
```

Comment:
"Despite the constraints, most HyperSpy operations can be performed lazily. Important points to note are:"